theano/tensor/nnet/ctc_wrapper.c

```diff
@@ -0,0 +1,242 @@
+#section support_code
+
+int ctc_check_result(ctcStatus_t retcode, const char * msg, int * status)
+{
+    if( CTC_STATUS_SUCCESS != retcode )
+    {
+        // Get error message from underlying library
+        const char * ctc_msg = ctcGetStatusString( retcode );
+
+        PyErr_Format( PyExc_RuntimeError,
+                      "%s | CTC library error message: %s",
+                      msg,
+                      ctc_msg );
+        *status = 1;
+    }
+    *status = 0;
+}
+
+void create_contiguous_input_lengths( PyArrayObject * input_lengths_arr,
+    int ** input_lengths )
+{
+    int num_elements = PyArray_DIMS( input_lengths_arr )[0];
+
+    *input_lengths = (int *) malloc( num_elements * sizeof(int) );
+
+    if ( NULL == (*input_lengths) )
+        return;
+
+    for( int elem_idx = 0; elem_idx < num_elements; ++elem_idx )
+    {
+        (*input_lengths)[elem_idx] = *( (int *) PyArray_GETPTR1( input_lengths_arr, elem_idx ) );
+    }
+}
+
+void create_flat_labels( PyArrayObject * label_matrix, int ** flat_labels, 
+    int ** label_lengths )
+{
+    int rows = PyArray_DIMS( label_matrix )[0];
+    int cols = PyArray_DIMS( label_matrix )[1];
+
+    *flat_labels = (int *) malloc( rows * cols * sizeof(int) );
+    if ( NULL == (*flat_labels) )
+        return;
+
+    *label_lengths = (int *) malloc( rows * sizeof(int) );
+    if ( NULL == (*label_lengths) )
+    {
+        free( *flat_labels );
+        *flat_labels = NULL;
+        return;
+    }
+
+    int label_index = 0;
+    for( int row_idx = 0; row_idx < rows; ++row_idx )
+    {
+        int label_length = 0;
+        for( int col_idx = 0; col_idx < cols; ++col_idx )
+        {
+            int label = *( (int *) PyArray_GETPTR2( label_matrix, row_idx, col_idx ) );
+            if ( label >= 0 )  // negative values are assumed to be padding
+            {
+                (*flat_labels)[ label_index++ ] = label;
+                ++label_length;
+            }
+        }
+        (*label_lengths)[ row_idx ] = label_length;
+    }
+}
+
+#section support_code_apply
+
+int APPLY_SPECIFIC(ctc_cost_cpu)(PyArrayObject *  in_activations,
+                                 PyArrayObject *  in_labels,
+                                 PyArrayObject *  in_input_lengths,
+                                 PyArrayObject ** out_costs,
+                                 PyArrayObject ** out_gradients)
```

Comment:
This can be fixed with the `_cop_num_outputs` attribute on the class.  Set it to 2 and the second output argument will be set to NULL if not provided on the node.